### PR TITLE
Create a merged figure for co-occurrence and disease distribution of mutations

### PR DESCRIPTION
**Pull request by jashapiro**
#### What scientific question is your analysis addressing?

This is along the path to creating a final figure for https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/images/figure_sketches/occurrence-fig-sketch.jpg

#### What was your approach?

I had already created the two subfigures, so this PR merges them into a single image, with some changes to formatting to make sure they are aligned and labels are appropriately placed.

#### Which areas should receive a particularly close look?

Is it clear how the figure is produced, in case it needs modification later?

#### Is there anything that you want to discuss further?

How does the figure look? Any suggestions for formatting changes? @dvenprasad had suggested adding ticks below the x axis labels on the cooccurence plot, which I have not yet implemented, but should be easily done.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

yes

#### What types of results are included (e.g., table, figure)?

Figures: the most relevant one here is `plots/combined_top50.png`

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
@kgaonkar6 and I are going to take a look at this together tomorrow morning.

**Comment by jashapiro** [comment]
> Hi @jashapiro looks like README needs a minor edit
> `sh analyses/interaction-plots/01-create-interaction-plots.sh` errors out
> `bash analyses/interaction-plots/01-create-interaction-plots.sh` works perfectly

Oh, good catch!

File: analyses/interaction-plots/scripts/03-plot_interactions.R [hunk 1]
```diff
@@ -184,28 +193,88 @@ disease_df <- disease_df %>%
...
+      angle = -90,
+      hjust = 1,
+      vjust = 0.5
+    ),
+    axis.title.y = element_text(
+      vjust = -10 # keep the label close when combined
+    )
+)
+
+# Patchwork black magic
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/interaction-plots/scripts/03-plot_interactions.R]: Which part of "black magic" is confusing? 🤣 

I will update this.

File: analyses/interaction-plots/scripts/03-plot_interactions.R [hunk 2]
```diff
@@ -184,28 +199,93 @@ disease_df <- disease_df %>%
...
+    axis.title.y = element_text(
+      vjust = -10 # keep the label close when combined
+    )
+  )
+
+# Combine plots with <patchwork>
+# Layout of the two plots will be one over the other (1 column), 
+# with the upper plot 3/4 the height of the lower plot
+combined_plot <- disease_plot2 + cooccur_plot2 +
+  plot_layout(ncol = 1, heights = c(3, 4)) &
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/03-plot_interactions.R]: I like the new heights 👍

File: analyses/interaction-plots/scripts/03-plot_interactions.R [hunk 3]
```diff
@@ -184,28 +199,93 @@ disease_df <- disease_df %>%
...
     x = "",
     y = "Samples with mutations",
     fill = "Diagnosis"
   ) + 
   colorblindr::scale_fill_OkabeIto() + 
+  scale_x_discrete(
+    limits = xscale2,
+    breaks = disease_df$gene
+  ) + 
+  scale_y_continuous(expand = c(0, 0.5, 0.1, 0))+
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/03-plot_interactions.R]:
```suggestion
  scale_y_continuous(expand = c(0, 0.5, 0.1, 0)) + 
```

**Comment by jharenza** [comment]
@jashapiro these look nice! Aesthetically, would you consider making your x-axis of the co-occurrance/samples with mutations plot having gene names going from bottom to top instead of top to bottom? I think it is more readable that way, especially when looking at the barplot above. Just a suggestion! Glad to see expected results as well!

**Comment by jashapiro** [comment]
> @jashapiro these look nice! Aesthetically, would you consider making your x-axis of the co-occurrance/samples with mutations plot having gene names going from bottom to top instead of top to bottom? I think it is more readable that way, especially when looking at the barplot above. Just a suggestion! Glad to see expected results as well!

I've tried it both ways, and I think I preferred this one. For the barplot alone, that is definitely better (along with a 45° angle), but when sticking them together, I found this to be the least confusing (for me). If there are desires to change it as we get to final final figures, it is an easy change.

File: Dockerfile [hunk 4]
```diff
@@ -303,6 +303,16 @@ RUN pip3 install "cython==0.29.15" && \
...
+# Packages required for rna-seq-composition
+RUN apt-get update -qq && apt-get -y --no-install-recommends install \
+    && install2.r --error \
+    --deps TRUE \
+    EnvStats \
+    janitor
+
+# Patchwork for plot compositions
+RUN R -e "devtools::install_github('thomasp85/patchwork', ref = 'c67c6603ba59dd46899f17197f9858bc5672e9f4')"
+
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply Dockerfile]: Yeah, I think that is a merge error. I only put one in myself.

File: analyses/interaction-plots/scripts/03-plot_interactions.R [hunk 5]
```diff
@@ -64,15 +65,22 @@ option_list <- list(
...
 )
 
 # Parse options
 opts <- parse_args(OptionParser(option_list = option_list))
 
 if (!is.na(opts$disease_table)){
-  if (is.na(opts$disease_plot)){
-    stop("If disease_table is specified, disease_plot  must also be specified")
+  if (is.na(opts$disease_plot) | is.na(opts$combined_plot)){
+    stop("If disease_table is specified, disease_plot and/or combined plot must also be specified")
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/interaction-plots/scripts/03-plot_interactions.R]: I think my logic is correct... It isn't an XOR, which is how people often interpret a bare "or" in text.

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/interaction-plots/scripts/03-plot_interactions.R]: 👍 I agree with the logic

**Pull request merged**